packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -104,6 +104,20 @@ export function usePasteHandler( props ) {
 			}
 
 			const files = [ ...getFilesFromDataTransfer( clipboardData ) ];
+			const isInternal = clipboardData.getData( 'rich-text' ) === 'true';
+
+			// If the data comes from a rich text instance, we can directly use it
+			// without filtering the data. The filters are only meant for externally
+			// pasted content and remove inline styles.
+			if ( isInternal ) {
+				const pastedValue = create( {
+					html,
+					preserveWhiteSpace,
+				} );
+				addActiveFormats( pastedValue, value.activeFormats );
+				onChange( insert( value, pastedValue ) );
+				return;
+			}
```

Comment:
This change here introduced a regression. Basically if you have a plain link in your paragraph for instance, and you cut the link and paste it over some text to create a real link, the link won't be created.

Reply:
This is just reverting https://github.com/WordPress/gutenberg/pull/48254/files#diff-11720f5193f80b98003f574708df7d6107ff4f286b6a5c31f66dce1ef73b5123L154, but rebased with the multi line stuff removed 🤔

Reply:
I think the regression is caused by the change in order here #55195